pkg/dao/repository.go

```diff
@@ -34,10 +31,52 @@ func (r repositoryDaoImpl) Create(newRepo api.CreateRepository) error {
 		return result.Error
 	}
 	if err := db.DB.Create(&newRepoConfig).Error; err != nil {
-		if err.Error() == uniqueConstraintErrMsg {
-			return fmt.Errorf("repository with this URL already belongs to organization")
+		if isUniqueViolation(err) {
+			return &Error{BadValidation: true, Message: "Repository with this URL already belongs to organization "}
 		}
 		return err
 	}
 	return nil
 }
+
+func (r repositoryDaoImpl) Fetch(orgId string, uuid string) api.Repository {
```

Comment:
Is this PR also going to add a GET for /repositories/:uuid?

Reply:
added!